### PR TITLE
Dobble Command initialization

### PR DESCRIPTION
**Pull request by Kmaschta**
My first commit at Marmelab, the epic one !

It's a simple symfony command with one argument, you can try it with `php application.php dobble:run <element_per_card>`
For now I check if argument is a positive integer and print it.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "require": {
+        "symfony/symfony": "^2.7"
```
Comment (fzaninotto) [inline composer.json]: You should require only the Symfony `Console` component - no need to grab the full-stack framework.

File: src/Dobble/DobbleCommand.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
```
Comment (fzaninotto) [inline src/Dobble/DobbleCommand.php:1]: php formatting is weird. Run [php-cs-fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) on your code to see the usual coding standards (refs PSR/1). It'd be a good idea to add php-cs-fixer to your Travis build, too (e.g. https://github.com/fzaninotto/Faker/blob/master/Makefile#L4-L6)

File: src/Dobble/DobbleCommand.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+class DobbleCommand extends Command {
+	protected function configure () {
+		$this->setName('dobble:run');
+		$this->setDescription('Command-line card generator for dobble-like game');
+
+		$this->addArgument(
+			'element_per_card',
+			InputArgument::REQUIRED,
+			'Number of element per card'
```
Comment (fzaninotto) [inline src/Dobble/DobbleCommand.php]: s/element/elements

File: src/Dobble/DobbleCommand.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+		$nb_element = $input->getArgument('element_per_card');
+
+		if($nb_element <= 0) {
+			$output->writeln('<error>Number of element must be a postive integer</error>');
+			return 0;
+		}
+
+		$output->writeln('Element per card: <info>' . (string)$nb_element . '</info>');
+	}
+}
```
Comment (fzaninotto) [inline src/Dobble/DobbleCommand.php]: You must configure your text editor to add a newline at the end of the file - that's a convention.

File: src/Dobble/DobbleCommand.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		$nb_element = $input->getArgument('element_per_card');
+
+		if($nb_element <= 0) {
+			$output->writeln('<error>Number of element must be a postive integer</error>');
+			return 0;
+		}
+
+		$output->writeln('Element per card: <info>' . (string)$nb_element . '</info>');
```
Comment (fzaninotto) [inline src/Dobble/DobbleCommand.php]: use `sprintf()` instead of concatenation.

File: src/application.php [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+require __DIR__.'/../vendor/autoload.php';
+require __DIR__.'/Dobble/DobbleCommand.php';
+
+use Dobble\DobbleCommand;
+use Symfony\Component\Console\Application;
+
+$app = new Application();
+$app->add(new DobbleCommand());
+$app->run();
```
Comment (fzaninotto) [inline src/application.php]: newline at the end of the file

**Comment by fzaninotto** [comment]
Please update the README right away, explaining how to install and use your project. Also, add a `.travis.yml` file with a basic (passing) test to ensure that the CI configuration is correct.

**Comment by Kmaschta** [comment]
All misc fix applied and travis passing, it run a simple `exit 0` on make test.
Unit testing for next commit.

**Comment by fzaninotto** [comment]
Excellent. Please take the habit to prefix your PR with either:
- '[WIP]' (work in progress): the PR isn't finished yet, but the review can start
- '[RFR]' (ready for review): the PR is considered finished; if the reviewer is OK, they can merge directly

I merge even though you have to configure your text editor for newlines at the end of file.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+language: php
+php:
+    - 7
+before_script: make install
+script: make test
```
Comment (jpetitcolas) [inline .travis.yml:5]: Missing end line

File: composer.json [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
+    "require": {
```
Comment (jpetitcolas) [inline composer.json:2]: Miss the autoload part (see [symfony-standard configuration](https://github.com/symfony/symfony-standard/blob/2.8/composer.json) for a reference), project name, type, license, and PHP required version.

File: makefile [hunk 9]
```diff
@@ -0,0 +1,5 @@
+install:
+	composer install --prefer-source
+
+test:
+	exit 0
```
Comment (jpetitcolas) [inline makefile:5]: Missing end line (but I love this `make test` however :p).

File: src/Dobble/DobbleCommand.php [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $nb_elements = $input->getArgument('elements_per_card');
+
+        if ($nb_elements <= 0) {
+            $output->writeln('<error>Number of elements must be a postive integer</error>');
+
+            return 0;
```
Comment (jpetitcolas) [inline src/Dobble/DobbleCommand.php:31]: Return 1 for an error. Or better, try to throw an exception.

File: src/application.php [hunk 11]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env php
+<?php
+
+require __DIR__.'/../vendor/autoload.php';
+require __DIR__.'/Dobble/DobbleCommand.php';
```
Comment (jpetitcolas) [inline src/application.php:5]: Should not be required once autoload is well configured.

**Pull request merged**